### PR TITLE
Added Q-tensor example

### PR DESCRIPTION
**Pull request by joshichaitanya3**
Added a morpho example to minimize the Q-tensor free energy subject to anchoring boundary conditions. Also added a write-up (PDF and .tex) detailing the physics and morpho implementation.

### CODE REVIEW

File: examples/qtensor/qtensor.morpho [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+}
+
+// Define anchoring energy at the boundary
+fn anchoring(x, q) {
+  var t = tangent()
+  var wxx = t[0]*t[0]-0.5
+  var wxy = t[0]*t[1]
+  return (q[0]-wxx)^2+(q[1]-wxy)^2
+}
+
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:47]: // Define parts of the energy?

File: examples/qtensor/qtensor.morpho [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+var bulk = AreaIntegral(landau, q_tensor)
+var anchor = LineIntegral(anchoring, q_tensor)
+var elastic = GradSq(q_tensor)
+
+var problem = OptimizationProblem(m)
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:52]: //Define the Optimization problem on the mesh

File: examples/qtensor/qtensor.morpho [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+problem.addenergy(bulk)
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:53]: //Add the energies to the optimization problem

File: examples/qtensor/qtensor.morpho [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+  
+    // Show(plotselection(m, srefine, grade=2))
+  
+    // Create a mesh refiner
+    var mr=MeshRefiner([m, q_tensor, bnd])
+  
+    // Perform the refinement
+    var refmap = mr.refine(selection=srefine)
+  
+    // Show(plotmesh(m, grade=1))
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:81]: Why is this commented out?

File: examples/qtensor/qtensor.morpho [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+problem.addenergy(elastic, prefactor = K)
+problem.addenergy(anchor, selection=bnd, prefactor = EA)
+
+var opt = FieldOptimizer(problem, q_tensor)
+// Minimize the free energy!
+opt.linesearch(500)
+
+
+// Refine the mesh based on the elastic energy density/
+// This for loop can be commented out
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:63]: Why can this for loop be commented out? If you're telling people to comment this out why have the refine_adaptively flag?

File: examples/qtensor/qtensor.morpho [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+// Refine the mesh based on the elastic energy density/
+// This for loop can be commented out 
+if (refine_adaptively){
+  for (i in 1..4){
+  
+    // Select elements that have a large contribution to the energy
+    var en = elastic.integrand(q_tensor)
+    var mean = en.sum()/en.count()
+    var srefine = Selection(m) // Start with an empty selection
+    for (id in 0...en.count()) if (en[0,id]>1.5*mean) srefine[2,id]=true // Identify high energy elements
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:71]: What does "high energy" mean here (I see that is elements that are over %150 of the mean but you should say that)

File: examples/qtensor/qtensor.morpho [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+  
+    // Show(plotselection(m, srefine, grade=2))
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:73]: Commented out code?

File: examples/qtensor/qtensor.morpho [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+  
+    // Now refinement is done update the problems and optimizers
+    for (el in [problem, opt]) el.update(refmap)
+  
+    // Use the new mesh and field
+    m = refmap[m]
+    q_tensor = refmap[q_tensor]
+    bnd = refmap[bnd]
+  
+    equiangulate(m)
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:91]: Whats this do?

File: examples/qtensor/qtensor.morpho [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+    // Show(plotmesh(m, grade=1))
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:92]: Commented out code

File: examples/qtensor/qtensor.morpho [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+  
+    opt.linesearch(500)
+  }
+}
+// Define some helper functions to plot the result
+  
+fn sign(x) {
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:99]: We should add this as a builtin function (created #44 )

File: examples/qtensor/qtensor.morpho [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+  if (x<0.0){
+    return -1.0
+  } else if (x>0.0){
+    return 1.0
+  } else{
+  return 0.0
+  }
+}
+
+fn qtodirector(q) {
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:109]: add Short description

File: examples/qtensor/qtensor.morpho [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+  var g = Graphics()
+  for (i in 0...nv) {
+    var x = v.column(i)
+    g.display(Cylinder(x-nn[i]*dl, x+nn[i]*dl, aspectratio=0.3))
+  }
+  return g
+}
+
+// Visualize the result
+// var g=plotmesh(m, grade=1)
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:144]: Commented out code

File: examples/qtensor/qtensor.morpho [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+var splot = plotfield(S, style="interpolate")
+var gnn=visualize(m, nn, 0.05)
+var gdisp = splot+gnn
+// Show(gdisp)
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:148]: Commented out code

File: examples/qtensor/qtensor.morpho [hunk 14]
```diff
@@ -0,0 +1,152 @@
...
+  var nv = v.dimensions()[1]
+  var g = Graphics()
+  for (i in 0...nv) {
+    var x = v.column(i)
+    g.display(Cylinder(x-nn[i]*dl, x+nn[i]*dl, aspectratio=0.3))
+  }
+  return g
+}
+
+// Visualize the result
```
Comment (ConduitDan) [inline examples/qtensor/qtensor.morpho:143]: Add comments for this block saying what each line does

**Comment by softmattertheory** [comment]
I really like this, @joshichaitanya3; I think the documentation could be a good model for the other examples and we can merge this into the manual at some point.

**Review by ConduitDan — CHANGES_REQUESTED** [review]
Hi @joshichaitanya3, 
This example need a little some clean up if its going to be user facing. 

In general this needs:

-  An intro section briefly explaining what problem script what what it is a good example of. (you should mention this using mesh refinement)
-  More explicit comments 
-  Additionally the PDF attached seems unfinished. But @softmattertheory we should come up with a workflow for adding documents for examples via just their .tex and having them built automatically

@joshichaitanya3 If you want to talk about any of this I'm free this afternoon or tomorrow afternoon.

**Comment by ConduitDan** [comment]
Look like this just got merged in. I've created #45 to track the changes I mentioned above.

**Comment by softmattertheory** [comment]
Oops, mea culpa, @ConduitDan ! I should note that the other examples don't necessarily meet these (excellent requirements!) Maybe we should have a convo about the above; it would be nice to figure out a standard template for examples etc. I feel that documentation belongs in the manual rather than the examples directory... but I really like the above.

**Pull request merged**